aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -3,66 +3,76 @@ module AptosFramework::Stake {
     /// Validator Config not published.
-    const EVALIDATOR_CONFIG: u64 = 4;
+    const EVALIDATOR_CONFIG: u64 = 3;
     /// Not enough stake to join validator set.
-    const ESTAKE_TOO_LOW: u64 = 5;
+    const ESTAKE_TOO_LOW: u64 = 4;
     /// Too much stake to join validator set.
-    const ESTAKE_TOO_HIGH: u64 = 6;
+    const ESTAKE_TOO_HIGH: u64 = 5;
     /// Account is already a validator or pending validator.
-    const EALREADY_VALIDATOR: u64 = 7;
+    const EALREADY_VALIDATOR: u64 = 6;
     /// Account is not a validator.
-    const ENOT_VALIDATOR: u64 = 8;
+    const ENOT_VALIDATOR: u64 = 7;
     /// Can't remove last validator.
-    const ELAST_VALIDATOR: u64 = 9;
-    /// Delegation from the address already exists in this pool
-    const EDELEGATION_ALREADY_EXIST: u64 = 10;
-    /// Delegation to this stake pool will make it exceed tht maximum stake specified in ValidatorSet.
-    const EDELEGATION_EXCEED_MAX: u64 = 11;
-
-    /// Basic unit of stake delegation, it's stored in StakePool.
-    struct Delegation has store {
-        coins: Coin<TestCoin>,
-        rewards: Coin<TestCoin>,
-        locked_until_secs: u64,
+    const ELAST_VALIDATOR: u64 = 8;
+    /// Total stake exceeds maximum allowed.
+    const ESTAKE_EXCEEDS_MAX: u64 = 9;
+    /// Account is already registered as a validator candidate.
+    const EALREADY_REGISTERED: u64 = 10;
+    /// Account does not have the right ownership capability.
+    const ENOT_OWNER: u64 = 11;
+    /// No coins in inactive state to withdraw from specified pool.
+    const ENO_COINS_TO_WITHDRAW: u64 = 12;
+    /// Account does not have the right operator capability.
+    const ENOT_OPERATOR: u64 = 13;
+
+    /// Capability that represents ownership and can be used to control the validator and the associated stake pool.
+    /// Having this be separate from the signer for the account that the validator resources are hosted at allows
+    /// modules to have control over a validator.
+    struct OwnerCapability has key, store {
+        pool_address: address,
     }
 
-    /// Aggregation of delegation and represent a validator's voting power, stored in ValidatorInfo.
-    /// Invariants:
-    /// 1. voting_power = sum(active + pending_inactive)
-    /// 2. user interact with pending_active and inactive if it's in the ValidatorSet.
-    /// 3. user interact with active, inactive if it's not in the ValidatorSet.
-    /// 4. pending_active and pending_inactive are empty if it's not in the ValidatorSet.
-    struct StakePool has key, store {
-        // sum of active and pending_inactive stakes, updated on epoch boundary for active validators.
-        voting_power: u64,
-        // sum of active and pending_active stakes, updated when delegate/unlock happens.
-        next_epoch_voting_power: u64,
+    /// Each validator has a separate StakePool resource.
+    ///
+    /// Similar to validator set, any changes to stake for an active validator only take effect in the next epoch.
+    ///
```

Comment:
We should probably define the different states for stake before defining what the current and next rounds stake will be. The following could also be broken down into a few categories:

Changes in stake for an active validator:
...

Changes in stake for an inactive validator:
...

5 bullets in a row with loosely related context isn't clean.

Reply:
nit

Reply:
Updated. Sorry I missed this comment earlier.